sp1/program/src/main.rs

```diff
@@ -0,0 +1,15 @@
+#![no_main]
+sp1_zkvm::entrypoint!(main);
+
+use std::str::FromStr;
+
+use sp1_zkvm::io::{read, write};
+
+use bitcoin::BtcBlocks;
+use lc::prove_btc_blocks;
+
+fn main() {
+    let blocks = BtcBlocks::from_str(&read::<String>()).expect("to unwrap btc blocks");
```

Comment:
Since this is in both sp1 and risc0, might be worth putting this into the lc lib itself, and have that accept as its input a string.